autogluon/utils/tabular/ml/models/abstract/abstract_model.py

```diff
@@ -58,7 +59,7 @@ class AbstractModel:
     model_info_name = 'info.pkl'
     model_info_json_name = 'info.json'
 
-    def __init__(self, path: str, name: str, problem_type: str, objective_func, num_classes=None, stopping_metric=None, model=None, hyperparameters=None, features=None, feature_types_metadata=None, debug=0, **kwargs):
+    def __init__(self, path: str, name: str, problem_type: str, objective_func, num_classes=None, stopping_metric=None, model=None, hyperparameters=None, features=None, feature_types_metadata: FeatureTypesMetadata = None, debug=0, **kwargs):
```

Comment:
It would be good to add brief documentation about `FeatureTypesMetadata`, since this class is important for users to understand in order to add a new model.

Reply:
Agreed, added. Also improved documentation on some of the other arguments.